src/main/java/net/minestom/server/crypto/ArgumentSignatures.java

```diff
@@ -1,34 +1,34 @@
 package net.minestom.server.crypto;
 
-import net.minestom.server.utils.binary.BinaryReader;
-import net.minestom.server.utils.binary.BinaryWriter;
-import net.minestom.server.utils.binary.Writeable;
+import net.minestom.server.network.NetworkBuffer;
 import org.jetbrains.annotations.NotNull;
 
 import java.util.List;
 
-public record ArgumentSignatures(@NotNull List<@NotNull Entry> entries) implements Writeable {
+import static net.minestom.server.network.NetworkBuffer.STRING;
+
+public record ArgumentSignatures(@NotNull List<@NotNull Entry> entries) implements NetworkBuffer.Writer {
     public ArgumentSignatures {
         entries = List.copyOf(entries);
     }
 
-    public ArgumentSignatures(BinaryReader reader) {
-        this(reader.readVarIntList(Entry::new));
+    public ArgumentSignatures(@NotNull NetworkBuffer reader) {
+        this(reader.readCollection(Entry::new));
     }
 
     @Override
-    public void write(@NotNull BinaryWriter writer) {
-        writer.writeVarIntList(entries, BinaryWriter::write);
+    public void write(@NotNull NetworkBuffer writer) {
+        writer.writeCollection(entries);
     }
 
-    public record Entry(@NotNull String name, @NotNull MessageSignature signature) implements Writeable {
-        public Entry(BinaryReader reader) {
-            this(reader.readSizedString(), new MessageSignature(reader));
```

Comment:
Doesn't the indirection via Type reduce performance?

Reply:
In theory yes, but in practice it is very minimal. And having the ability to describe a writable structure using constants could allow some interesting optimizations.

Though performance isn't the main goal here. This version is much more maintainable.